python/cudf_polars/cudf_polars/containers/column.py

```diff
@@ -45,10 +45,13 @@ def _dtype_short_repr_to_dtype(dtype_str: str) -> pl.DataType:
     if dtype_str.startswith("list["):
         stripped = dtype_str.removeprefix("list[").removesuffix("]")
         return pl.List(_dtype_short_repr_to_dtype(stripped))
-    pl_type = pl.datatypes.convert.dtype_short_repr_to_dtype(dtype_str)
+    pl_type = polars.datatypes.convert.dtype_short_repr_to_dtype(dtype_str)
     if pl_type is None:
         raise ValueError(f"{dtype_str} was not able to be parsed by Polars.")
-    return pl_type() if inspect.isclass(pl_type) else pl_type
+    if isinstance(pl_type, polars.datatypes.DataTypeClass):
+        return pl_type()
+    else:
+        return pl_type
```

Comment:
Adding some tests for this. Apparently `pl.Datetime` can hit this branch while most everything else hits the other branch.